numpy/core/_methods.py

```diff
@@ -165,7 +165,8 @@ def _mean(a, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
     is_float16_result = False
 
     rcount = _count_reduce_items(arr, axis, keepdims=keepdims, where=where)
-    if rcount == 0 if where is True else umr_any(rcount == 0):
+    if (where is True and rcount == 0) or \
```

Comment:
I don't get this logic - surely `_count_reduce_items` can return an nd array even if `where is True`, and this will still break?

Reply:
I think it's impossible for `_count_reduce_items` to return ndarray with `where=True`. It returns a scalar `items = nt.intp(1)` with accumulated items `items *= ...`. Have you got an example where it returns an array when `where=True`?

Reply:
Ah you're right, it doesn't do what I think it does.

Reply:
Yeah, it was confusing for me 😅. Should I revert it to original `if ... if .. else:`?

Reply:
Maybe make a temporary variable with a nice name if you like to avoid the two `if` on a single line?

Reply:
On the second thought I revered it to original notation as it's all clear to me, sorry for this confusion. All pushed.